streamline/docs/cdk-usage.md

```diff
@@ -0,0 +1,160 @@
+# CDK Usage Guide - ACM for Nitro Enclaves Streamline
+
+This guide covers advanced deployment scenarios using [CDK CLI](https://docs.aws.amazon.com/cdk/v2/guide/cli.html) directly and detailed configuration options.
+
+## Configuration Interface
+```typescript
+interface NitroEnclavesAcmStreamlineConfig {
+  certificateConfig: {
+    certificateName?: string;
+    domainName: string;
+    isPrivate: boolean;
+    hostedZoneId?: string;
+    validationType?: 'DNS' | 'EMAIL';
+    pcaArn?: string;
+    existingCertificateArn?: string;
+  };
+  roleConfig?: {
+    roleName?: string;
+  };
+  instanceConfig: {
+    instanceName?: string;
+    keyPairName: string;
+    serverType: 'NGINX' | 'APACHE';
+    amiType: 'AL2' | 'AL2023';
+    instanceType: string;
+  };
+  region: string;
+  account: string;
+}
+```
+The default configuration file can be found in [`src/config/default-config.ts`](../src/config/default-config.ts)
+
+## CDK Deployment Examples:
+### 1. Private Certificate Deployment:
+Change the configuration in the [`src/config/default-config.ts`](../src/config/default-config.ts) file.
+```typescript
+const config: NitroEnclavesAcmStreamlineConfig = {
+  certificateConfig: {
+    certificateName: 'PrivateAcmneCertificate',
+    domainName: 'private.example.com',
+    isPrivate: true,
+    pcaArn: 'arn:aws:acm-pca:my-region-1:123456789:certificate-authority/xxx-yyyy'
+  },
+  roleConfig: {
+    roleName: 'PrivateAcmneRole'
+  },
+  instanceConfig: {
+    instanceName: 'PrivateAcmneInstance',
+    keyPairName: 'my-private-key',
+    serverType: 'NGINX',
+    amiType: 'AL2',
+    instanceType: 'c5.xlarge'
+  },
+  region: 'my-region-1',
+  account: '123456789'
+};
+```
+
+Deploy the stacks
+
+```bash
+cdk deploy --all --require-approval never
```

Comment:
I do not think the default should have the example with the `--require-approval never` flag set as a general note.

While this guide is aimed at advanced users, it might be taken verbatim by people without as much experience. An advanced user will know if they want to suppress the need for approvals, but inexperienced users might make security-relevant changes without learning about it.

Same goes for all the instances of this below.